### ISSUE TITLE
[plando bug] Extra triforce pieces added to the item pool instead of junk when ice traps are disabled

### ISSUE THREAD

**Issue by ETR-BTF** [TestRunnerSRL/OoT-Randomizer#1444]
The attached plando specifies that there are 100 pieces in the pool, but if you run it with ice traps off then 6 extra get added. If you then specify 7 gold rupees in the item pool (which is 1 more than exist in the Balanced item pool, on which this was rolled) you only get 5 extra pieces. 4 extra pieces for 8 gold rupees etcetera until you eventually go back to the intended 100 pieces. I suspect it might have been caused by #1236, tho I'll admit I don't understand the PR well enough to be 100% certain. The issue also exists on v6.0.26, the version in which the aforementioned PR was merged, but on older versions it fails to generate, presumably due to the bug that was fixed in that PR. 

What should happen is that extra junk is added, but instead 6 extra Triforce Pieces are added, and adding surplus junk manually (e.g. 7+ gold rupees) will mitigate/resolve this.

I could not reproduce this without a plando file, which is why I believe it's a plando bug.

[Triforce_Pieces_on_all_GS.txt](https://github.com/TestRunnerSRL/OoT-Randomizer/files/7572418/Triforce_Pieces_on_all_GS.txt)

**Comment by Star1468** [comment]
Adding freestanding rupees/recovery hearts and pot drops to the shuffle seems to have worsened the problem (https://github.com/rrealmuto/OoT-Randomizer/tree/pots). The attached plando specifies 168 pieces, but the seeds generated on the pot shuffle branch will generate between 212-216 pieces, well more than specified.
[OoT_D369C_CMEEYD55WI_Spoiler.txt](https://github.com/TestRunnerSRL/OoT-Randomizer/files/7966799/OoT_D369C_CMEEYD55WI_Spoiler.txt)
[Plando.txt](https://github.com/TestRunnerSRL/OoT-Randomizer/files/7966800/Plando.txt)

**Comment by fenhl** [comment]
Still happens in current versions of the randomizer.

<details>
<summary>Updated plando</summary>

```json
{
    "settings": {
        "tokensanity": "all",
        "triforce_hunt": true,
        "triforce_count_per_world": 100,
        "triforce_goal_per_world": 100,
        "junk_ice_traps": "off",
        "starting_items": {
            "Gold Skulltula Token": 100
        }
    },
    "item_pool": {
        "Gold Skulltula Token": 0,
        "Triforce Piece": 100
    },
    "locations": {
        "Deku Tree GS Compass Room": "Triforce Piece",
        "Deku Tree GS Basement Vines": "Triforce Piece",
        "Deku Tree GS Basement Gate": "Triforce Piece",
        "Deku Tree GS Basement Back Room": "Triforce Piece",
        "Dodongos Cavern GS Side Room Near Lower Lizalfos": "Triforce Piece",
        "Dodongos Cavern GS Scarecrow": "Triforce Piece",
        "Dodongos Cavern GS Alcove Above Stairs": "Triforce Piece",
        "Dodongos Cavern GS Vines Above Stairs": "Triforce Piece",
        "Dodongos Cavern GS Back Room": "Triforce Piece",
        "Jabu Jabus Belly GS Water Switch Room": "Triforce Piece",
        "Jabu Jabus Belly GS Lobby Basement Lower": "Triforce Piece",
        "Jabu Jabus Belly GS Lobby Basement Upper": "Triforce Piece",
        "Jabu Jabus Belly GS Near Boss": "Triforce Piece",
        "Bottom of the Well GS West Inner Room": "Triforce Piece",
        "Bottom of the Well GS East Inner Room": "Triforce Piece",
        "Bottom of the Well GS Like Like Cage": "Triforce Piece",
        "Ice Cavern GS Spinning Scythe Room": "Triforce Piece",
        "Ice Cavern GS Heart Piece Room": "Triforce Piece",
        "Ice Cavern GS Push Block Room": "Triforce Piece",
        "Forest Temple GS First Room": "Triforce Piece",
        "Forest Temple GS Lobby": "Triforce Piece",
        "Forest Temple GS Raised Island Courtyard": "Triforce Piece",
        "Forest Temple GS Level Island Courtyard": "Triforce Piece",
        "Forest Temple GS Basement": "Triforce Piece",
        "Fire Temple GS Boss Key Loop": "Triforce Piece",
        "Fire Temple GS Song of Time Room": "Triforce Piece",
        "Fire Temple GS Boulder Maze": "Triforce Piece",
        "Fire Temple GS Scarecrow Climb": "Triforce Piece",
        "Fire Temple GS Scarecrow Top": "Triforce Piece",
        "Water Temple GS Behind Gate": "Triforce Piece",
        "Water Temple GS Near Boss Key Chest": "Triforce Piece",
        "Water Temple GS Central Pillar": "Triforce Piece",
        "Water Temple GS Falling Platform Room": "Triforce Piece",
        "Water Temple GS River": "Triforce Piece",
        "Spirit Temple GS Metal Fence": "Triforce Piece",
        "Spirit Temple GS Sun on Floor Room": "Triforce Piece",
        "Spirit Temple GS Hall After Sun Block Room": "Triforce Piece",
        "Spirit Temple GS Lobby": "Triforce Piece",
        "Spirit Temple GS Boulder Room": "Triforce Piece",
        "Shadow Temple GS Invisible Blades Room": "Triforce Piece",
        "Shadow Temple GS Falling Spikes Room": "Triforce Piece",
        "Shadow Temple GS Single Giant Pot": "Triforce Piece",
        "Shadow Temple GS Near Ship": "Triforce Piece",
        "Shadow Temple GS Triple Giant Pot": "Triforce Piece",
        "KF GS Know It All House": "Triforce Piece",
        "KF GS Bean Patch": "Triforce Piece",
        "KF GS House of Twins": "Triforce Piece",
        "LW GS Bean Patch Near Bridge": "Triforce Piece",
        "LW GS Bean Patch Near Theater": "Triforce Piece",
        "LW GS Above Theater": "Triforce Piece",
        "SFM GS": "Triforce Piece",
        "HF GS Cow Grotto": "Triforce Piece",
        "HF GS Near Kak Grotto": "Triforce Piece",
        "Market GS Guard House": "Triforce Piece",
        "HC GS Tree": "Triforce Piece",
        "HC GS Storms Grotto": "Triforce Piece",
        "LLR GS House Window": "Triforce Piece",
        "LLR GS Tree": "Triforce Piece",
        "LLR GS Rain Shed": "Triforce Piece",
        "LLR GS Back Wall": "Triforce Piece",
        "Kak GS Tree": "Triforce Piece",
        "Kak GS Near Gate Guard": "Triforce Piece",
        "Kak GS Watchtower": "Triforce Piece",
        "Kak GS Skulltula House": "Triforce Piece",
        "Kak GS House Under Construction": "Triforce Piece",
        "Kak GS Above Impas House": "Triforce Piece",
        "Graveyard GS Bean Patch": "Triforce Piece",
        "Graveyard GS Wall": "Triforce Piece",
        "DMT GS Near Kak": "Triforce Piece",
        "DMT GS Bean Patch": "Triforce Piece",
        "DMT GS Above Dodongos Cavern": "Triforce Piece",
        "DMT GS Falling Rocks Path": "Triforce Piece",
        "GC GS Center Platform": "Triforce Piece",
        "GC GS Boulder Maze": "Triforce Piece",
        "DMC GS Crate": "Triforce Piece",
        "DMC GS Bean Patch": "Triforce Piece",
        "ZR GS Tree": "Triforce Piece",
        "ZR GS Ladder": "Triforce Piece",
        "ZR GS Near Raised Grottos": "Triforce Piece",
        "ZR GS Above Bridge": "Triforce Piece",
        "ZD GS Frozen Waterfall": "Triforce Piece",
        "ZF GS Above the Log": "Triforce Piece",
        "ZF GS Tree": "Triforce Piece",
        "ZF GS Hidden Cave": "Triforce Piece",
        "LH GS Bean Patch": "Triforce Piece",
        "LH GS Lab Wall": "Triforce Piece",
        "LH GS Small Island": "Triforce Piece",
        "LH GS Lab Crate": "Triforce Piece",
        "LH GS Tree": "Triforce Piece",
        "GV GS Small Bridge": "Triforce Piece",
        "GV GS Bean Patch": "Triforce Piece",
        "GV GS Behind Tent": "Triforce Piece",
        "GV GS Pillar": "Triforce Piece",
        "GF GS Top Floor": "Triforce Piece",
        "GF GS Archery Range": "Triforce Piece",
        "Wasteland GS": "Triforce Piece",
        "Colossus GS Bean Patch": "Triforce Piece",
        "Colossus GS Tree": "Triforce Piece",
        "Colossus GS Hill": "Triforce Piece",
        "OGC GS": "Triforce Piece"
    }
}
```

</details>

**Comment by fenhl** [comment]
I've been looking at this in a debugger and I'm still not quite sure what's going on, the plando code and item pool code are calling each other in a pattern that's hard to follow. Current best guess is that the plando code expects `pending_junk_pool` to consist entirely of junk items that are safe to duplicate, but the item pool code places extra major items (in this case Triforce pieces) there as well.
